### ISSUE TITLE
useless indirection in REST API

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#1607]
---

Author Name: **Guillaume Rousse** (@guillomovitch)
Original Redmine Issue: 2860, http://forge.fusioninventory.org/issues/2860
Original Date: 2015-01-13
Original Assignee: David Durieux

---

When requesting initial schedule, trough getConfig action, it is perfectly useless to send task specification to the agent, when there is no such task configured, ie:

Agent request for schedule:
GET '/plugins/fusioninventory/?action=getConfig&machineid=agent_id&task[ESX]=2.2.1&task[Deploy]=1.0'
Server answer:
{"configValidityPeriod":600,"schedule":[{"periodicity":3600,"delayStartup":3600,"task":"ESX","remote":"http:\/\/my.server\/plugins\/fusioninventory\/b\/esx\/"},{"task":"Deploy","remote":"http:\/\/my.server\/plugins\/fusioninventory\/b\/deploy\/"}]}

Agent request for ESX jobs:
GET '/plugins/fusioninventory/b/esx/?action=getJobs&machineid=agent_id'
Server answer:
{"jobs":[]}

Agent request for Deploy jobs:
GET '/plugins/fusioninventory/b/deploy/?action=getJobs&machineid=agent_id'
Server answer:
{"jobs":[]}

Basically, the initial answer should only contains actual tasks, defined for this agent, not potential tasks.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Durieux** (@ddurieux)
Original Date: 2015-02-19T21:44:03Z

---

Applied in changeset commit:185e38939f45632eaaabf86fbc9258f5d8c01ef7.
